File: lib-fast4Android-core/src/main/java/io/github/keep2iron/android/net/interceptor/AddCookiesInterceptor.java
/*
 * Create bt Keep2iron on 17-5-25 下午2:33
 * Copyright (c) 2017. All rights reserved.
 */

package io.github.keep2iron.android.net.interceptor;

import java.io.IOException;
import java.util.HashSet;

import io.github.keep2iron.android.util.SPUtils;
import okhttp3.Interceptor;
import okhttp3.Request;
import okhttp3.Response;

/**
 * This interceptor put all the Cookies in Pref in the Request.
 * Your implementation on how to get the Pref MAY VARY.
 * <p>
 * Created by tsuharesu on 4/1/15.
 */
public class AddCookiesInterceptor implements Interceptor {

    @Override
    public Response intercept(Chain chain) throws IOException {
        Request request = chain.request();
        Request.Builder builder = request.newBuilder();

        HashSet<String> preferences = (HashSet<String>) SPUtils.Companion.getInstance().getStringSet("cookies", new HashSet<>());
        for (String cookie : preferences) {
            builder.addHeader("Cookie", cookie);
        }

        return chain.proceed(builder.build());
    }
}

File: lib-network/src/main/java/io/github/keep2iron/network/interceptor/AddCookiesInterceptor.kt
/*
 * Create bt Keep2iron on 17-5-25 下午2:33
 * Copyright (c) 2017. All rights reserved.
 */

package io.github.keep2iron.network.interceptor

import java.io.IOException
import java.util.HashSet
import okhttp3.Interceptor
import okhttp3.Request
import okhttp3.Response

/**
 * This interceptor put all the Cookies in Pref in the Request.
 * Your implementation on how to get the Pref MAY VARY.
 *
 * 用于添加Cookie信息的拦截器
 *
 * Created by tsuharesu on 4/1/15.
 */
class AddCookiesInterceptor : Interceptor {

    @Throws(IOException::class)
    override fun intercept(chain: Interceptor.Chain): Response {
        val request = chain.request()
        val builder = request.newBuilder()

//        val preferences = SPUtils.instance.getStringSet("cookies", HashSet()) as HashSet<String>
//        for (cookie in preferences) {
//            builder.addHeader("Cookie", cookie)
//        }

        return chain.proceed(builder.build())
    }
}